pinot-core/src/main/java/org/apache/pinot/core/operator/filter/SortedInvertedIndexBasedFilterOperator.java

```diff
@@ -78,82 +79,92 @@ protected FilterBlock getNextBlock() {
     int[] dictIds = exclusive ? _predicateEvaluator.getNonMatchingDictIds() : _predicateEvaluator.getMatchingDictIds();
 
     if (0 < dictIds.length) {
-      // Sort the dictIds in ascending order, so that their respective ranges are adjacent if their dictIds are adjacent
-      Arrays.sort(dictIds);
+      if (_predicateEvaluator instanceof RangePredicateEvaluatorFactory.OfflineDictionaryBasedRangePredicateEvaluator) {
+        // optimized path for RANGE predicate evaluation using sorted index.
+        // dictIds will already be sorted since we use sorted index based evaluation
+        // for RANGE predicates only on offline segments.
+        // so build a pair with startDocId and endDocId
+        IntPair pair1 = invertedIndex.getDocIds(dictIds[0]);
```

Comment:
This is not efficient. We should add getStartDictId() and getEndDictId() to OfflineDictionaryBasedRangePredicateEvaluator, and directly get the start and end dict id.

Reply:
Good suggestion. Done... no need to call getMatchingDictIds